src/Propel/Generator/Model/ForeignKey.php

```diff
@@ -466,7 +466,8 @@ public function getForeignTableName(): ?string
         }
 
         $database = $this->getDatabase();
-        if ($database && ($schema = $this->parentTable->guessSchemaName()) && $platform && $platform->supportsSchemas()) {
+        $schema = $this->parentTable->guessSchemaName();
```

Comment:
Umph, test fail because apparently `$database && ($schema = $this->parentTable->guessSchemaName())` tests if parent table has a database, so that `Table::guessSchemaName()` does not fail with an exception:
```php
    public function guessSchemaName(): ?string
    {
        return $this->schema ?: $this->database->getSchema();
    }
```
Pretty horrible. I think it should be
```php
    public function guessSchemaName(): ?string
    {
        if($this->schema){
            return $this->schema;
        }
        return $this->database ? $this->database->getSchema() : null;
    }
```